src/structures/emoji.ts

```diff
@@ -48,3 +66,11 @@ export class Emoji extends Base {
     if (data.user !== undefined) this.user = new User(this.client, data.user)
   }
 }
+
+/** https://discord.com/developers/docs/resources/emoji#modify-guild-emoji-json-params */
+export interface ModifyGuildEmojiParams {
+  /** Name of the emoji */
+  name: string
+  /** Roles to which this emoji will be whitelisted */
+  roles: string[] | null
```

Comment:
What do you think about allowing user to put `Role | Role[]` for `roles` too? And making all props optional here will make sense.